amd/src/pxaiendpoint.js

```diff
@@ -1,171 +1,195 @@
-define(['jquery', "core/ajax", "core/str", 'core/templates', 'core/modal_factory', "core/modal_events"], function ($, Ajax, Str, Templates, ModalFactory, ModalEvents) {
-    var EventCreator = function (id, cmid, contextid, steps_data, assignmentid, attempt_text) {
-
-        this.instanceid = id;
-        this.cmid = cmid;
-        this.contextId = contextid;
-        this.steps = steps_data;
+import $ from "jquery";
+import Ajax from "core/ajax";
+import AISubmissionEvent from "./app/ai/event";
+import AISubmissionPageChangeEvent from "./app/ai/page_change_event";
+
+/**
+ * @param {number} id
+ * @param {number} cmid
+ * @param {number} contextid
+ * @param {string} steps_data
+ * @param {number} assignmentid
+ * @param {string} attempt_text
+ */
+export const init = (
+    id,
+    cmid,
+    contextid,
+    steps_data,
+    assignmentid,
+    attempt_text
+) => {
+
+    let EventCreator = function (stepsData, assignmentId, attemptText) {
         this.currStep = 0;
         this.selStart = 0;
         this.selEnd = 0;
-        this.assignmentid = assignmentid
-        this.attempt_text = attempt_text
-
+        this.assignmentid = assignmentId;
+        this.attempt_text = attemptText;
         this.init();
-
-    }
+    };
 
     EventCreator.prototype.init = function () {
 
         this.setAttemtCountText(this.attempt_text);
+        const container = $(".actions-container");
+
+        /**
+         * @param {HTMLElement} button
+         */
+        const highlightStepButton = (button) => {
+            if (!(button instanceof HTMLElement)) {
+                return;
+            }
+            button.classList.add('btn-outline-primary');
+            button.classList.remove('btn-outline-secondary');
+        };
+
+        /**
+         * @param {HTMLElement} button
+         */
+        const blurStepButton = (button) => {
+            if (!(button instanceof HTMLElement)) {
+                return;
+            }
+            button.classList.remove('btn-outline-primary');
+            button.classList.add('btn-outline-secondary');
+        };
 
-        var getSelectedTextareaText = function (name) {
+        let getSelectedTextareaText = function (name) {
 
-            var selText = '';
-            var txtarea = document.getElementsByName(name);
-            txtarea = (txtarea && txtarea.length > 0) ? txtarea[0] : null;
-            if (txtarea) {
-                var start = txtarea.selectionStart;
+            let selText = '';
+            let textArea = document.getElementsByName(name);
+            textArea = (textArea && textArea.length > 0) ? textArea[0] : null;
```

Comment:
I would suggest just using null coalescence here for simplicity
`let textArea = document.getElementsByName(name)[0] ?? null;`

Reply:
Javascript changes will be in the next commit from the other issue.
Because the whole Javascript implementation looks really bad... :disappointed: